apps/bfd-server/bfd-server-launcher/src/test/java/gov/cms/bfd/server/launcher/DataServerLauncherAppIT.java

```diff
@@ -29,6 +34,102 @@
 public final class DataServerLauncherAppIT {
   /** The POSIX signal number for the <code>SIGTERM</code> signal. */
   private static final int SIGTERM = 15;
+  /** Regex for access log entries */
+  private static final String accessLogPattern =
+      new StringJoiner(" ")
+          .add("^(\\S+)") // Address or Hostname
+          .add("\\S+") // Dash symbol (-) separator
+          .add("\"([^\"]*)\"") // Remote user info
+          .add("\\[([^\\]]+)\\]") // Request timestamp
+          .add("\"([A-Z]+) ([^ \"]+) HTTP\\/[0-9.]+\"") // First line of request
+          .add("\"([^ \"]+)\"") // Request query string
+          .add("([0-9]{3})") // Response status
+          .add("([0-9]+|-)") // Bytes transferred
+          .add("([0-9]+|-)") // Time taken to serve request
+          .add("(\\S+)") // BlueButton-OriginalQueryId
+          .add("([0-9]+|-)") // BlueButton-OriginalQueryCounter
+          .add("\\[([^\\]]+)\\]") // BlueButton-OriginalQueryTimestamp
+          .add("([0-9]+|-)") // BlueButton-DeveloperId
+          .add("\"([^\"]*)\"") // BlueButton-Developer
+          .add("([0-9]+|-)") // BlueButton-ApplicationId
+          .add("\"([^\"]*)\"") // BlueButton-Application
+          .add("([0-9]+|-)") // BlueButton-UserId
+          .add("\"([^\"]*)\"") // BlueButton-User
+          .add("(\\S+)") // BlueButton-BeneficiaryId
+          .toString();
+
+  /**
+   * Verifies the regex for valdiating our access log entries adequately avoids edge cases that
+   * could break our alerts, which depend on logs
+   */
+  @Test
+  public void checkAccessLogFormat() {
```

Comment:
We discussed this some and the other side of this is that this regex only exists for the IT test that uses it to verify output (and that must be an IT test because it has to start the server process and generate a log file). So putting that regex in another class and then putting the unit tests for the regex in yet another class is a substantial increase in scope and indirection. Trading that complexity for having a unit test in the IT class seemed reasonable to me.

Reply:
Hm I see the place that uses this regex to parse the logs is external to BFD, so this is kind of a weird case in terms of testing. Worried we need to remember to update this when/if we change the pattern for parsing the logs, since the place we use pattern this won't break this test if we change it. Not sure how to connect the two, but this at least offers some coverage, so I'm cool with it.